core/src/main/java/org/mule/processor/AbstractNonBlockingMessageProcessor.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.processor;
+
+import org.mule.NonBlockingVoidMuleEvent;
+import org.mule.api.MessagingException;
+import org.mule.api.MuleEvent;
+import org.mule.api.MuleException;
+import org.mule.api.CompletionHandler;
+import org.mule.api.exception.MessagingExceptionHandler;
+import org.mule.api.exception.MessagingExceptionHandlerAware;
+import org.mule.api.transport.ReplyToHandler;
+
+/**
+ * Abstract implementation of {@link org.mule.processor.NonBlockingMessageProcessor} that determines if processing should
+ * be performed blocking or non-blocking and defines two template methods for each mode of processing.
```

Comment:
"and defines two template methods for each mode of processing" is kind of redundant